patroni/dcs/etcd.py

```diff
@@ -215,6 +215,13 @@ def get_etcd_client(config):
                 sleep(5)
         return client
 
+    def set_ttl(self, ttl):
+        ttl = int(ttl)
+        if self._ttl != ttl:
+            self.reset_cluster()
```

Comment:
This is a trick to update TTL of leader and member keys in etcd. Update is happening only when `update_leader` and `touch_member` are called. I don't want to call them explicitly, but want to run next iteration of ha loop immediately.

`set_ttl` is called from `Patroni.reload_config`, after ha_loop, but before `schedule_next_run`, which calls `dcs.watch`

`watch` method is implemented is such way that it will not "watch" for "leader" key if cluster object is None, but will wait for event `AbstractDCS.watch` method. Line 222 is taking care about generating this event.